### ISSUE TITLE
Unable to osc pull / osc commit to fix project with broken link: missing sha256 checksum

### ISSUE THREAD

**Issue by krig** [openSUSE/open-build-service#4188]
Issue/Feature description
====================
I have a project on OBS with a broken link (change committed to devel project while previous revision was accepted to openSUSE Factory).

The first issue is that committing new revisions to a devel project while a submission is in progress to openSUSE:Factory breaks the link: This shouldn't happen, in my opinion.

The second issue is that repairing the link fails.

Expected result
============
The following sequence of commands should repair the link:

1. `osc pull`
2. `[fix the merge conflict in cluster-glue.spec]`
3. `osc resolved cluster-glue.spec`
4. `osc commit`

How to Reproduce
===============
1. `osc co network:ha-clustering:Factory cluster-glue`
2. `osc pull`
3. `vim cluster-glue.spec`
4. `osc resolved cluster-glue.spec`
5. `osc commit -m "fixed the merge conflict"`

At the final step, the following happens:

```
$ osc commit -m "fix spec"
Unknown license 'GPL-2.0 and LGPL-2.1+'
- package has baselibs.conf: (unchanged)
Sending    cluster-glue.spec
Transmitting file data There is no sha256 sum for file _service.
This could be due to an outdated working copy.
Please update your working copy with osc update and
commit again afterwards.
```

Further information
===============
* Broken repository: https://build.opensuse.org/package/show/network:ha-clustering:Factory/cluster-glue

**Comment by krig** [comment]
Fixed by updating `osc` to `osc-0.162.1-228.1`.

**Comment by marcus-h** [comment]
On 2017-12-05 07:22:31 -0800, Kristoffer Grönlund wrote:
> Issue/Feature description
> ====================
> I have a project on OBS with a broken link (change committed to devel project while previous revision was accepted to openSUSE Factory).
> 
> The first issue is that committing new revisions to a devel project while a submission is in progress to openSUSE:Factory breaks the link: This shouldn't happen, in my opinion.
> 
This is in some sense related to PR#337 [1]. See also the
discussion on the factory ml [2]. We should probably move this
discussion to the PR or the ml.

> ```
> $ osc commit -m "fix spec"
> Unknown license 'GPL-2.0 and LGPL-2.1+'
> - package has baselibs.conf: (unchanged)
> Sending    cluster-glue.spec
> Transmitting file data There is no sha256 sum for file _service.
> This could be due to an outdated working copy.
> Please update your working copy with osc update and
> commit again afterwards.
> ```
> 
This is fixed in commit f698103. Hence, I'm closing this issue.

[1] https://github.com/openSUSE/osc/pull/337
[2] https://lists.opensuse.org/opensuse-factory/2017-10/msg00262.html
